libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKPasscodeTextField.m

```diff
@@ -0,0 +1,156 @@
+ 
+ Copyright (c) 2018-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFSDKPasscodeTextField.h"
+#import "UIColor+SFSDKPasscodeView.h"
+#import "SFSDKPasscodeViewConfig.h"
+
+static NSUInteger   const kMaxPasscodeLength                 = 8;
+static CGFloat      const kDefaultPadding                    = 20.0f;
+static CGFloat      const kPasscodeCircleDiameter            = 24.f;
+static CGFloat      const kPasscodeCircleSpacing             = 16.f;
+
+@interface SFSDKPasscodeTextField()
+
+@property (nonatomic, strong) UIColor * fillColor;
+
+@end
+@implementation SFSDKPasscodeTextField
+
+- (instancetype)init {
+    if (self = [super init]) {
+        
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame andLength:(NSUInteger)length{
+    if (self = [super initWithFrame:frame]) {
+        SFSDKPasscodeViewConfig *config = [SFSDKPasscodeViewConfig createDefaultConfig];
+        
+        _passcodeLength = length;
+        _passcodeLengthKnown = (length != 0);
+        self.keyboardType = UIKeyboardTypeNumberPad;
+        self.backgroundColor = config.secondaryColor;
+        self.tintColor = [UIColor clearColor];
+        self.borderStyle = UITextBorderStyleNone;
+        self.layer.borderColor = config.borderColor.CGColor;
+        self.fillColor = config.primaryColor;
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame andLength:(NSUInteger)length andViewConfig:(SFSDKPasscodeViewConfig *)config{
```

Comment:
Reuse the overloaded initializer method and add the extra property assignments here.